### PR TITLE
feat(config): convenience class with static constants for referencing AWS managed rules

### PR DESCRIPTION
**Pull request by shivlaks**
follow-up with some feedback from the API review:
* added a class `ManagedRuleIdentifiers` with static constants for managed rule identifiers
* updated `README` and tests
* tried to use inclusive wording(blocked/ blocked list as opposed to blacklist)

we decided not to go with an enum like class as this is a breaking change that we don't
want to force on users as well as the limited value from wrapping a string around a class.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-config/lib/rule.ts [hunk 1]
```diff
@@ -390,6 +390,877 @@ export class CustomRule extends RuleNew {
   }
 }
 
+/**
+ * Managed rules that are supported by AWS Config.
+ * @see https://docs.aws.amazon.com/config/latest/developerguide/managed-rules-by-aws-config.html
+ */
+export class ManagedRuleIdentifier {
```
Comment (skinny85) [inline packages/@aws-cdk/aws-config/lib/rule.ts:397]: Now that we drifted away from the enum-like class here... Does it make sense to name this `ManagedRuleIdentifiers`?

**Review by skinny85 — APPROVED** [review]
Looks good. One suggestion that you can take or leave.

Added `do-not-merge`, but feel free to push this at your leisure.

File: packages/@aws-cdk/aws-config/lib/rule.ts (hunk 1 above)
Comment (shivlaks) [reply packages/@aws-cdk/aws-config/lib/rule.ts:397]: good call out. that seems sensible with this shift. updating!

Comment (shivlaks) [reply packages/@aws-cdk/aws-config/lib/rule.ts]: done.

**Pull request merged**